packages/vscode-extension/src/commands/ImportObjects.ts

```diff
@@ -0,0 +1,238 @@
+/*
+ ** Copyright (c) 2021 Oracle and/or its affiliates.  All rights reserved.
+ ** Licensed under the Universal Permissive License v 1.0 as shown at https://oss.oracle.com/licenses/upl.
+ */
+import * as fs from 'fs';
+import * as path from 'path';
+import { Uri, window } from 'vscode';
+import CustomObjectService from '../service/ImportObjectService';
+import { ANSWERS, ERRORS, IMPORT_OBJECTS } from '../service/TranslationKeys';
+import { actionResultStatus, ApplicationConstants, InteractiveAnswersValidator, ProjectInfoService } from '../util/ExtensionUtil';
+import BaseAction from './BaseAction';
+
+const objectTypes: {
+	name: string;
+	value: { name: string; type: string; prefix: string; hasRelatedFiles: boolean; relatedFiles?: { type: string }[] };
+}[] = require('@oracle/suitecloud-cli/src/metadata/ObjectTypesMetadata');
+
+const COMMAND_NAME = 'importobjects';
+
+export default class ImportObjects extends BaseAction {
+	private customObjectService: CustomObjectService;
+
+	constructor() {
+		super(COMMAND_NAME);
+		this.customObjectService = new CustomObjectService(this.messageService, this.translationService);
+	}
+
+	protected init(uri?: Uri) {
+		super.init(uri);
+		this.customObjectService.setVsConsoleLogger(this.vsConsoleLogger);
+	}
+
+	protected async execute() {
+		if (!this.activeFile) {
+			// Already checked in validate
+			return;
+		}
+		const projectInfoService = new ProjectInfoService(this.getProjectFolderPath());
+
+		const destinationFolder = this.getDestinationFolder(this.activeFile);
+		if (!destinationFolder.startsWith(ApplicationConstants.FOLDERS.OBJECTS)) {
+			this.messageService.showErrorMessage(this.translationService.getMessage(IMPORT_OBJECTS.ERROR.INCORRECT_FOLDER));
+			return;
+		}
+
+		let appId: string | undefined;
+		if (projectInfoService.isSuiteAppProject()) {
+			appId = await this.promptAppId(projectInfoService);
+			if (appId == undefined) {
+				this.messageService.showInformationMessage(this.translationService.getMessage(IMPORT_OBJECTS.PROCESS_CANCELED));
+				return;
+			}
+			if (appId == '') {
+				appId = undefined;
+			}
+		}
+
+		const selectedObjectTypes = await this.promptObjectTypes();
+
+		if (!selectedObjectTypes) {
+			this.messageService.showInformationMessage(this.translationService.getMessage(IMPORT_OBJECTS.PROCESS_CANCELED));
+			return;
+		}
+
+		const scriptId = await this.promptSelectedScriptId();
+
+		if (scriptId == undefined) {
+			this.messageService.showInformationMessage(this.translationService.getMessage(IMPORT_OBJECTS.PROCESS_CANCELED));
+			return;
+		}
+
+		const listObjectsResult = await this.listObjects(appId, selectedObjectTypes, scriptId);
```

Comment:
If  we change the reassignment of appId variable to undefined in line 54, the `listObjects` method should only allow string parameters (that could be empty strings) but never undefined, and the same in the service method (`ImportObjectsService.listobjects(...)`). 
Somehow I would like to follow a rule like, if the parameter was undefined the execution should stop, if you don't want to filter by that parameter let's make it's string value equals to an emptyString.
Following that approach we should not call the service with undefined values, and the method should not allow them to be undefined. I think it will make the code more robust.

Reply:
appId could be undefined if the project is an ACP

Reply:
If we are on ACP or we don't want to import the objects that belong to an specific suiteApp we can set it's value to an emptyString.

Reply:
Sorry David, I think this is adding some useless code (setting appid to emptystring) when listObjects could handle undefined appIds as it is doing right now.
I don't see so much benefits on this change.

Reply:
More important I would have to look for a condition in ImportObjectService (Line 55) to ensure that appId length is 0 to not add it to the arguments.
It's more easy to ensure just that is not undefined

Reply:
The condition it's already there in the service.
`if (appId) {commandArgs.appid = appId;}`
My point here is that we should ensure we are not passing any parameter with undefined value by error. That's why I would change the listObjects() signature in the ImportObjectService to not accept undefined parameters.